src/main/java/com/google/devtools/build/lib/bazel/repository/downloader/HttpDownloader.java

```diff
@@ -73,6 +77,17 @@ public void setMaxRetryTimeout(Duration maxRetryTimeout) {
     this.maxRetryTimeout = maxRetryTimeout;
   }
 
+  public void setMaxParallelDownloads(int maxParallelDownloads) {
+    if (maxParallelDownloads >= this.maxParallelDownloads) {
+      // increase the number of possible parallel downloads
+      semaphore.release(maxParallelDownloads - this.maxParallelDownloads);
```

Comment:
The docs for `Semaphore` make it clear that this use case (releasing beyond the initial number of permits) is supported, but since this isn't necessarily true for any semaphore implementation in other languages, it could use a comment.